packages/web3-utils/src/converters.ts

```diff
@@ -436,3 +437,56 @@ export const jsonInterfaceMethodToString = (
 
 	return `${json.name}(${flattenTypes(false, json.inputs).join(',')})`;
 };
+
+export const convertToValidType = (
+	value: string | number | bigint,
+	desiredType: ValidTypes,
+): ValidReturnTypes[ValidTypes] => {
+	switch (desiredType) {
+		case ValidTypes.HexString:
+			return toHex(value);
+		case ValidTypes.NumberString:
+			return hexToNumberString(toHex(value));
+		case ValidTypes.Number:
+			return toNumber(value);
+		case ValidTypes.BigInt:
+			return BigInt(toNumber(value));
+		default:
+			// TODO Replace with proper error
+			throw new Error('Invalid desiredType provided');
+	}
+};
+
+// TODO Handle nested objects
+export function convertObjectPropertiesToValidType<
+	// Object can have any properties. Unless specified as [key: string] index type didn't detect the record key correctly
+	// Also objects property types can be arrays, or other customs types as well so we have to specify any to cover all
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	ObjectType extends Record<any, any>,
+	Properties extends (keyof ObjectType)[],
+	ReturnType extends ValidTypes,
+>(
+	object: ObjectType,
+	convertibleProperties: Properties,
+	desiredType: ReturnType,
+): FormatValidReturnType<ObjectType, Properties, ReturnType> {
+	const convertedObject = { ...object } as FormatValidReturnType<
+		ObjectType,
+		Properties,
+		ReturnType
+	>;
+
+	for (const convertibleProperty of convertibleProperties) {
+		if (convertedObject[convertibleProperty] === undefined) continue;
+
+		// TODO Check why TS compiler is unable to detect the matching type of the deep properties
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-expect-error
+		convertedObject[convertibleProperty] = convertToValidType(
+			object[convertibleProperty],
+			desiredType,
+		);
+	}
+
+	return convertedObject;
+}
```

Comment:
@nazarhussain Helped me out with this, not sure how/when we should try cleaning this up

Reply:
Let's split this out as a separate issue, discuss its scope and then work on it. It's really complex to to handle the nested objects so we should do it in a separate PR.

Reply:
https://github.com/ChainSafe/web3.js/issues/4582